site/src/components/PaginationWidget/PaginationNavButton.tsx

```diff
@@ -0,0 +1,84 @@
+import { type ReactNode, useEffect, useState } from "react";
+
+import Button from "@mui/material/Button";
+import Tooltip from "@mui/material/Tooltip";
+import { useTheme } from "@emotion/react";
+
+type PaginationNavButtonProps = {
+  children: ReactNode;
+  disabled: boolean;
+  disabledMessage: ReactNode;
+  onClick: () => void;
+  "aria-label": string;
+
+  disabledMessageTimeout?: number;
+};
+
+export function PaginationNavButton({
+  children,
+  onClick,
+  disabled,
+  disabledMessage,
+  "aria-label": ariaLabel,
+  disabledMessageTimeout = 3000,
+}: PaginationNavButtonProps) {
+  const theme = useTheme();
+  const [showDisabledMessage, setShowDisabledMessage] = useState(false);
+
+  // Inline state sync - this is safe/recommended by the React team in this case
+  if (!disabled && showDisabledMessage) {
+    setShowDisabledMessage(false);
+  }
```

Comment:
I still don't like it 💀

Reply:
Yeah, I remember seeing an article about it called something like "the ugliest pattern in React" or something. It is better than using `useEffect` from a performance standpoint, and I guess also "philosophy" reasons (the React team is getting more bullish about how `useEffect` should be used strictly to sync with external systems, and if the state already exists inside React, you should do things like this)

Reply:
and yet we also have the wonderful `useSyncExternalStore` hook :lolsob:

Reply:
I mean, I get that the hook was mainly designed to help state management tools hook into React, but yeah. React's gonna React